src/main/java/com/shaft/cli/FileActions.java

```diff
@@ -681,10 +683,10 @@ private boolean isTargetOSUnixBased() {
             }
         } else {
             // remote execution
-            String targetOS = System.getProperty("targetOperatingSystem");
-            if ("Windows".equals(targetOS)) {
+            String targetOS = SHAFT.Properties.platform.targetPlatform();
+            if (Platform.WINDOWS.name().equals(targetOS)) {
                 return false;
-            } else if ("Linux".equals(targetOS) || "Mac".equals(targetOS)) {
+            } else if (Platform.LINUX.name().equals(targetOS) || Platform.MAC.name().equals(targetOS)) {
```

Comment:
note that we should no longer use the platform name as a hardcoded string @MahmoudElSharkawy @magdyheiba